lib/ProductOpener/Tags.pm

```diff
@@ -1702,6 +1704,188 @@ sub build_tags_taxonomy($$$) {
 }
 
 
+=head2 generate_tags_taxonomy_extract ( $tagtype, $tags_ref, $options_ref, $lcs_ref)
+
+Generate an extract of the taxonomy for a specific set of tags.
+
+=head3 Parameters
+
+=head4 tag type $tagtype
+
+=head4 reference to a list of tags ids $tags_ref
+
+=head4 reference to a hash of key/value options
+
+Options:
+- fields: comma separated lists of fields (e.g. "name,description,vegan:en,inherited:vegetarian:en" )
+
+Properties can be requested with their name (e.g."description") or name + a specific language (e.g. "vegan:en").
+Only properties directly defined for the entry are returned.
+To include inherited properties from parents, prefix the property with "inherited:" (e.g. "inherited:vegan:en").
+
+- include_parents: include entries for all direct parents of the requested tags
+- include_children: include entries for all direct children of the requested tags
+
+=head4 reference to an array of language codes
+
+Languages for which we want to extract names, synonyms, properties.
+
+=cut
+
+sub generate_tags_taxonomy_extract ($$$$) {
+
+	my $tagtype = shift;
+	my $tags_ref = shift;
+	my $options_ref = shift;
+	my $lcs_ref = shift;
+
+	# Return empty hash if the taxonomy does not exist
+	if (not defined $translations_to{$tagtype}) {
+		return {};
+	}
+
+	# For the options include_children or include_parents,
+	# we will need to include data for more tags that requested.
+	# @tags will hold the tags to include
+
+	my @tags = ();
+	my %requested_tags = ();
+	foreach my $tagid (@$tags_ref) {
+		push @tags, $tagid;
+		$requested_tags{$tagid} = 1;
+	}
+
+	my $fields_ref;
+	my @properties = ();
```

Comment:
Maybe it's normal in perl, but I'm a bit surprised that we use same name as `$properties`, I find it a bit confusing. We could also maybe make the intention more clear here by using a name like `@inherited_props`.

Reply:
Agreed.